api/cpp/cmake/SlintConfig.cmake.in

```diff
@@ -14,8 +14,41 @@ endif()
 add_library(@slint_cpp_impl@ @cmake_lib_type@ IMPORTED)
 set_target_properties(@slint_cpp_impl@ PROPERTIES @SLINT_LIB_PROPERTIES@)
 
-set(SLINT_COMPILER @SLINT_COMPILER@ CACHE FILEPATH "Path to the slint-compiler executable")
+function(_slint_download_compiler_and_cache)
+    set(SLINT_GITHUB_RELEASE @SLINT_VERSION@ CACHE STRING "GitHub Release to use for Slint Binary Artifact Downloads")
+
+    if (CMAKE_HOST_WIN32)
+        set(compiler_exe_suffix ".exe")
+        set(platform_suffix "windows")
+    else()
+        set(platform_suffix "${CMAKE_HOST_SYSTEM_NAME}-${CMAKE_HOST_SYSTEM_PROCESSOR}")
+    endif()
+
+    set(prebuilt_archive_filename "slint-compiler-${platform_suffix}.tar.gz")
+    set(download_url "https://github.com/slint-ui/slint/releases/download/${SLINT_GITHUB_RELEASE}/${prebuilt_archive_filename}")
+    set(download_directory "${CMAKE_BINARY_DIR}/slint-prebuilt")
+    message(STATUS "Downloading pre-built Slint compiler binary from ${download_url}")
+    file(DOWNLOAD "${download_url}" "${download_directory}/${prebuilt_archive_filename}" STATUS download_status)
+    list(GET download_status 0 download_code)
+    if (NOT download_code EQUAL 0)
+        list(GET download_status 1 download_message)
+        message(FATAL_ERROR "Download of Slint compiler binary package failed: ${download_message}")
+    else()
+        file(ARCHIVE_EXTRACT INPUT "${download_directory}/${prebuilt_archive_filename}" DESTINATION "${download_directory}")
+        file(REMOVE "${download_directory}/${prebuilt_archive_filename}")
+
+        set(SLINT_COMPILER "${download_directory}/slint-compiler${compiler_exe_suffix}")
+        set(SLINT_COMPILER "${SLINT_COMPILER}" CACHE STRING "Path to the slint-compiler executable" FORCE)
+        set(SLINT_COMPILER "${SLINT_COMPILER}" PARENT_SCOPE)
+    endif()
+endfunction()
+
+set(SLINT_COMPILER @SLINT_COMPILER@ CACHE STRING "Path to the slint-compiler executable")
```

Comment:
should the description documentation mention "download"? 
Should we actually document `SLINT_COMPILER` in our cmake docs?

Reply:
I'm not entirely sure yet if we should do so. But if we do, I'd rather do it in combination with a concrete use-case, one that's not internal to us.

For example if we were to document how to cross-compile for raspbian with C++ using only our binary packages. But then.... those would already be configured to download the compiler and the user doesn't have to set this variable.

If the user ends up building Slint from source, then none of this indirection is necessary as the rust build compiles the compiler just fine.